x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -307,20 +289,16 @@ private void ensureOpen() {
                     protected void doRun() {
                         for (SparseFileTracker.Gap gap : gaps) {
                             try {
-                                ensureOpen();
-                                if (readLock.tryLock() == false) {
-                                    throw new AlreadyClosedException("Cache file channel is being evicted, writing attempt cancelled");
+                                if (reference.tryIncRef() == false) {
+                                    throw new AlreadyClosedException("Cache file channel has been released and closed");
```

Comment:
Related to your discussion below I believe we could add:
```
assert false : "expected a non-closed channel reference"
```

Reply:
Yea that seems like a safe thing to try :) thanks!